### PR TITLE
implement queue timing using Datadog profiler, remove JFR implementation

### PR DESCRIPTION
**Pull request by richardstartin**
# What Does This Do

# Motivation

# Additional Notes

### CODE REVIEW

File: dd-java-agent/agent-profiling/profiling-ddprof/src/main/java/com/datadog/profiling/ddprof/QueueTimeTracker.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  @Override
+  public void setScheduler(Class<?> scheduler) {
+    this.scheduler = scheduler;
+  }
+
+  @Override
+  public void close() {
+    long endMillis = System.currentTimeMillis();
```
Comment (jbachorik) [inline dd-java-agent/agent-profiling/profiling-ddprof/src/main/java/com/datadog/profiling/ddprof/QueueTimeTracker.java:42]: We are ok with `System.currentTimeMillis()` not being strictly monotonic?

File: dd-java-agent/agent-profiling/profiling-ddprof/src/main/java/com/datadog/profiling/ddprof/DatadogProfiler.java [hunk 2]
```diff
@@ -437,4 +446,11 @@ public void recordSetting(String name, String value, String unit) {
       profiler.recordSetting(name, value, unit);
     }
   }
+
+  public QueueTimeTracker newQueueTimeTracker(long localRootSpanId, long spanId) {
+    if (profiler != null) {
+      return new QueueTimeTracker(profiler, queueTimeThreshold, localRootSpanId, spanId);
+    }
+    return null;
+  }
```
Comment (jbachorik) [inline dd-java-agent/agent-profiling/profiling-ddprof/src/main/java/com/datadog/profiling/ddprof/DatadogProfiler.java:455]: This is not used anywhere, right?

File: dd-java-agent/agent-profiling/profiling-controller-ddprof/src/main/java/com/datadog/profiling/controller/ddprof/DatadogProfilerTimer.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  @Override
+  public Timing start(TimerType type) {
+    long localRootSpanId = 0;
+    long spanId = 0;
+    AgentSpan activeSpan = AgentTracer.activeSpan();
+    if (activeSpan != null) {
+      spanId = activeSpan.getSpanId();
+      AgentSpan rootSpan = activeSpan.getLocalRootSpan();
+      localRootSpanId = rootSpan == null ? spanId : rootSpan.getSpanId();
+    }
```
[the remark this answers is not in the document]
Comment (richardstartin) [reply dd-java-agent/agent-profiling/profiling-controller-ddprof/src/main/java/com/datadog/profiling/controller/ddprof/DatadogProfilerTimer.java:31]: I thought about this too. I introduced the `TimerType` to allow future extensibility, but don't envisage using this extensibility in the near future. So it doesn't matter for now.

File: dd-java-agent/agent-profiling/profiling-ddprof/src/main/java/com/datadog/profiling/ddprof/DatadogProfiler.java (hunk 2 above)
Comment (richardstartin) [reply dd-java-agent/agent-profiling/profiling-ddprof/src/main/java/com/datadog/profiling/ddprof/DatadogProfiler.java:455]: It is used by the Timer implementation

Comment (jbachorik) [reply dd-java-agent/agent-profiling/profiling-ddprof/src/main/java/com/datadog/profiling/ddprof/DatadogProfiler.java:455]: 🤦  GH search was not very helpful ...

**Review by jbachorik — APPROVED** [review]

**Pull request merged**